actor/v7action/application_summary_test.go

```diff
@@ -574,6 +573,115 @@ var _ = Describe("Application Summary Actions", func() {
 						)
 					})
 
+					When("getting application deployment succeeds", func() {
+						When("the deployment is active", func() {
+							When("the deployment strategy is rolling", func() {
+								When("the deployment is in progress", func() {
+									BeforeEach(func() {
+										fakeCloudControllerClient.GetDeploymentsReturns(
+											[]resources.Deployment{
+												{
+													GUID:         "some-deployment-guid",
+													Strategy:     "rolling",
+													StatusValue:  "ACTIVE",
+													StatusReason: "DEPLOYING",
+												},
+											},
+											nil,
+											nil,
+										)
+									})
+									It("returns the deployment information", func() {
+										Expect(summary.Deployment).To(Equal(resources.Deployment{
+											GUID:         "some-deployment-guid",
+											Strategy:     "rolling",
+											StatusValue:  "ACTIVE",
+											StatusReason: "DEPLOYING",
+										}))
+									})
+								})
+
+								When("the deployment is canceled", func() {
+									When("the deployment is in progress", func() {
+										BeforeEach(func() {
+											fakeCloudControllerClient.GetDeploymentsReturns(
+												[]resources.Deployment{
+													{
+														GUID:         "some-deployment-guid",
+														Strategy:     "rolling",
+														StatusValue:  "ACTIVE",
+														StatusReason: "CANCELLING",
+													},
+												},
+												nil,
+												nil,
+											)
+										})
+										It("returns the deployment information", func() {
+											Expect(summary.Deployment).To(Equal(resources.Deployment{
+												GUID:         "some-deployment-guid",
+												Strategy:     "rolling",
+												StatusValue:  "ACTIVE",
+												StatusReason: "CANCELLING",
+											}))
+										})
+									})
+								})
+							})
+						})
+
+						When("the deployment is finalized", func() {
+							BeforeEach(func() {
+								fakeCloudControllerClient.GetDeploymentsReturns(
+									[]resources.Deployment{
+										{
+											GUID:         "",
+											Strategy:     "",
+											StatusValue:  "",
+											StatusReason: "",
+										},
+									},
+									nil,
+									nil,
+								)
+							})
+							It("returns no deployment information", func() {
+								Expect(summary.Deployment).To(Equal(resources.Deployment{
+									GUID:         "",
+									Strategy:     "",
+									StatusValue:  "",
+									StatusReason: "",
+								}))
+							})
+						})
+					})
+
+					When("getting application deployment fails", func() {
+						BeforeEach(func() {
+							fakeCloudControllerClient.GetDeploymentsReturns(
+								nil,
+								ccv3.Warnings{"get-deployments-warning"},
+								errors.New("some-error"),
+							)
+						})
+
+						It("returns the warnings and error", func() {
+							Expect(executeErr).To(MatchError("some-error"))
+							Expect(warnings).To(ConsistOf(
```

Comment:
Yes, it is because of when the test is happening (after all other function stubs are invoked) as we saw earlier.

Reply:
I get that these warnings are all *returned* in this scenario. But IMHO they're not all *relevant* to this scenario, and therefore should probably not be enumerated.

If some logic changes for one of the other warnings we'd need to rewrite this assertion to match. The test becomes more robust — and more expressive — by limiting the assertion to checking for just the presence of the single, relevant warning and not the exhaustive contents of the full array.

Reply:
I agree. We discussed this while working on these tests and moved it around to suit the process when we were pairing. If something changes for these warning, most of the tests in this will need to change.